auth/package.json

```diff
@@ -32,6 +32,7 @@
     },
     "devDependencies": {
         "@azure/core-auth": "^1.4.0",
+        "@azure/identity": "^4.2.0",
```

Comment:
I don't see this package being used

Reply:
https://github.com/microsoft/vscode-azuretools/pull/1723/files#diff-9c30a2d74591058fcf9d06dbc7074281071ad805af0619e472a83df969ac8284R202

That lazy load tho~~

Reply:
Note about the new dependencies-- they were already included in the `@azure/ms-rest-azure-env` but I didn't want to import implicitly, so just included them in the package.json. It shouldn't be introducing new dependencies.